rpc/src/v1/impls/eth_pubsub.rs

```diff
@@ -251,18 +251,16 @@ impl<C: Send + Sync + 'static> EthPubSub for EthPubSubClient<C> {
 		kind: pubsub::Kind,
 		params: Trailing<pubsub::Params>,
 	) {
+
 		let error = match (kind, params.into()) {
-			(pubsub::Kind::NewHeads, None) => {
+			(pubsub::Kind::NewHeads, _) => {
```

Comment:
I don't like loosening it completely. I think the RPC and the spec should be as specific as possible, we'll run into plenty of issues if we allow leniency of parameters (like developers wondering why a parameter doesn't work if they make a typo or the parameter is not supported).

IMHO The spec just not got properly updated after `newBlocks` has been renamed to `newHeads`, `includeTransactions` paramteter in examples doesn't make much sense in case of headers-only subscription.

Having said that, I'm ok with allowing:
`eth_subscribe('newHeads')` and `eth_subscribe('newHeads', {})` call (optional options argument that we might add in the future), but not `eth_subscribe('newHeads', {'anything': true})`.

@tjade273 what do you think?

Reply:
Makes sense, I'll tweak it